File: web3rpc/rpc-specs/code-samples/python/debug/vMTracing/traceBlockByHash.py
from web3 import Web3
from web3py_ext import extend

host = "https://api.baobab.klaytn.net:8651"

blockHash = "0xed110b330152df2022d40fa3c38987643034aa56fc96079fb6c67b66a6ed4f19"

w3 = Web3(Web3.HTTPProvider(host))
debug_response = w3.debug.trace_block_by_hash(blockHash)

print(debug_response)

File: web3rpc/rpc-specs/code-samples/javascript/debug/vMTracing/traceBlockByHash.js
const { JsonRpcProvider } = require("@klaytn/ethers-ext");
(() => {
  const provider = new JsonRpcProvider("https://public-en-baobab.klaytn.net");
  const blockHash =
    "0xed110b330152df2022d40fa3c38987643034aa56fc96079fb6c67b66a6ed4f19";

  provider.debug
    .traceBlockByHash(blockHash, {}, (err, data, response) => {})
    .then((data) => {
      console.log(data);
    });
})();
